src/wallet/walletutil.cpp

```diff
@@ -57,35 +57,59 @@ static bool IsBerkeleyBtree(const fs::path& path)
         if (ec) {
             LogPrintf("%s: %s %s\n", __func__, ec.message(), it->path().string());
             continue;
         }
 
-        // Get wallet path relative to walletdir by removing walletdir from the wallet path.
-        // This can be replaced by boost::filesystem::lexically_relative once boost is bumped to 1.60.
-        const fs::path path = it->path().string().substr(offset);
-
-        if (it->status().type() == fs::directory_file && IsBerkeleyBtree(it->path() / "wallet.dat")) {
-            // Found a directory which contains wallet.dat btree file, add it as a wallet.
-            paths.emplace_back(path);
-        } else if (it.level() == 0 && it->symlink_status().type() == fs::regular_file && IsBerkeleyBtree(it->path())) {
-            if (it->path().filename() == "wallet.dat") {
-                // Found top-level wallet.dat btree file, add top level directory ""
-                // as a wallet.
-                paths.emplace_back();
-            } else {
-                // Found top-level btree file not called wallet.dat. Current bitcoin
-                // software will never create these files but will allow them to be
-                // opened in a shared database environment for backwards compatibility.
-                // Add it to the list of available wallets.
-                paths.emplace_back(path);
+        // We dont want to iterate trough those special node dirs
+        if (std::count(ignore_paths.begin(), ignore_paths.end(), it->path())) {
+            it.disable_recursion_pending();
+            continue;
+        }
+
+        try {
+            if (it->status().type() == fs::directory_file && IsBerkeleyBtree(it->path() / "wallet.dat")) {
+                // Found a directory which contains wallet.dat btree file, add it as a wallet.
+                // Get wallet path relative to walletdir by removing walletdir from the wallet path.
+                // This can be replaced by boost::filesystem::lexically_relative once boost is bumped to 1.60.
+                paths.emplace_back(it->path().string().substr(offset));
+            } else if (it.level() == 0 && it->symlink_status().type() == fs::regular_file && IsBerkeleyBtree(it->path())) {
+                if (it->path().filename() == "wallet.dat") {
+                    // Found top-level wallet.dat btree file, add top level directory ""
+                    // as a wallet.
+                    paths.emplace_back();
+                } else {
+                    // Found top-level btree file not called wallet.dat. Current bitcoin
+                    // software will never create these files but will allow them to be
+                    // opened in a shared database environment for backwards compatibility.
+                    // Add it to the list of available wallets.
+                    paths.emplace_back(it->path().filename());
+                }
             }
+        } catch (const std::exception& e) {
```

Comment:
I personally prefer "throwing" versions of the Boost.Filesystem API. But I don't think that combining "throwing" and "non-throwing" versions of the Boost.Filesystem API in the same function is a good design.

Reply:
with catch boost::exception
and  dynamic_cast<std.exception i run with a recursive path in an non catched     
 EXCEPTION: N5boost10filesystem16filesystem_errorE
so unless u have a hint, i leave it that way, for now?